libs/utils/commons/src/lib/castcle-name.ts

```diff
@@ -20,110 +20,20 @@
  * Thailand 10160, or visit www.castcle.com if you need additional information
  * or have any questions.
  */
-import {
-  BANNED_NAMES,
-  LENGTH_MAX,
-  LENGTH_MIN,
-  RANDOM_ADJECTIVE,
-  RANDOM_SUBJECTS,
-  RESERVE_NAMES,
-} from './configs/names';
 
-export class CastcleName {
-  slug: string;
-  isBanned: boolean;
-  isReserved: boolean;
-  isValid: boolean;
-  suggestCastcleId: string;
-  constructor(public name: string) {
-    this.slug = CastcleName.convertToSlug(name);
-    this.isBanned = CastcleName.isBannedName(name);
-    this.isReserved = CastcleName.isReserveName(name);
-    this.isValid = CastcleName.isValidName(name);
-    this.suggestCastcleId = CastcleName.suggestCastcleId(name);
-  }
-
-  /**
-   * Change name to lowercase + remove all space
-   * @param {string }name
-   * @returns {string}
-   */
-  static convertToSlug(name: string) {
-    return name
-      .toLowerCase()
-      .replace(/[^\w ]+/g, '')
-      .replace(/ +/g, '');
-  }
-
-  /**
-   * convert name to slug and check if it exist in configs/names/ban.names
-   * @param {string }name
-   * @returns {boolean} return true if the name is ban name
-   */
-  static isBannedName(name: string) {
-    return BANNED_NAMES.find((str) => str === this.convertToSlug(name))
-      ? true
-      : false;
-  }
-
-  /**
-   * convert name to slug and check if it exist in configs/names/reserve.names
-   * @param {string} name
-   * @returns {boolean}
-   */
-  static isReserveName(name: string) {
-    return RESERVE_NAMES.find((str) => str === this.convertToSlug(name))
-      ? true
-      : false;
-  }
+import { HttpService } from '@nestjs/axios';
+import { lastValueFrom, map } from 'rxjs';
 
-  /**
-   * convert name to slug and check if length in ranges(in config) and if it has any special character
-   * @param {string} name
-   * @returns {boolean} return true if name is valid
-   */
-  static isValidName(name: string) {
-    const lengthRule =
-      this.convertToSlug(name).length >= LENGTH_MIN &&
-      this.convertToSlug(name).length <= LENGTH_MAX;
-    const format = /[ `!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?~]/;
-    const specialCharacterRule = !format.test(name);
-    return (
-      lengthRule &&
-      specialCharacterRule &&
-      !this.isReserveName(name) &&
-      !this.isBannedName(name)
-    );
-  }
+export class CastcleName {
+  private static httpService = new HttpService();
 
-  /**
-   * remove all special character and if length is more than LENGTH_MAX slice it to the maxium length
-   * @param {string} name
-   * @returns {string}
-   */
-  static _preSuggest(name: string) {
-    //remove any special character
-    let replaceName = name.replace(
-      /[ `!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?~]/g,
-      '',
-    );
-    if (name.length >= LENGTH_MAX)
-      replaceName = replaceName.slice(0, LENGTH_MAX - 1);
-    return replaceName;
+  static toStug(keyword: string) {
+    return keyword?.toLowerCase().replace(/\s|\W/g, '');
```

Comment:
```suggestion
    return keyword?.toLowerCase().replace(/\W/g, '_');
```

1. `\s` = whitespace, tab, new line ส่วน `\W` = not A-Z,a-Z,0-9,_  ดังนั้นถ้าจะใช้ \W ก็ไม่ต้องมี \s ก็ได้ครับ
2. อยากให้ replace ด้วย _ ไม่งั้นสมมติ `james&luton` จะกลางเป็น `jamesluton` แล้วถ้า banned words มี slut อยู่ มันจะเจอเป็นคำหยาบ